Removing depricated `BitcoinHash` crate

dr-orlovsky opened the pull request:
New version of `rust-bitcoin` removed trait `BitcoinHash`. This is not yet got to the crates, but is required for Signet support in #239, so I am adding this as a separate PR to be edited & merged after the new bitcoin crate version will be released to crates.io

PS. this PR includes bulk of changes from #238, pls ignore them (w/o these changes it would not compile)

romanz commented:
LGTM, many thanks for the fix!
Will be happy to merge when rust-bitcoin 0.24 is released.

dr-orlovsky commented:
With the new bitcoin crate 0.24 release this PR is ready for review & merge. I have rebased it + also moved from direct usage of bitcoin_hashes crate towards version cross-imported from bitcoin directly: this helps to avoid version conflicts between local and bitcoin structures.

romanz commented:
Thanks!
